### DISCUSSION TITLE
Is there a good practice around src/tests? (and side question: what is your preferred virtual env?)

### DISCUSSION THREAD

**Discussion by Wabri** [Schroedinger-Hat/ImageGoNord-cli#24] [Q&A]
I've always try to find a good virtual environment manager for python (I'm not a py dev expert) and I remember that in the old days I used to use poetry. Today I taking the project back on maintaining and start reading the documentation https://python-poetry.org/docs/basic-usage/ and saw that the skeleton for the project is this:

```
poetry-demo
├── pyproject.toml
├── README.md
├── poetry_demo
│   └── __init__.py
└── tests
    └── __init__.py
```

Right now we have:

```
imagegonord-cli
├── README.md
├── src
│   └── __init__.py
└── tests
    └── __init__.py
```

is it better to change the src directory to the name of the project?

what virtual environment tool are you using for python?

**Comment by TheJoin95** [comment]
I was using virtual env and pip.
I like poetry approach, so we can go ahead with this implementation if you like

**Reply by Wabri** [reply]
I take the fork of @BugliL to help him in develop the new cli with the args, looking at the code is really impressive both readability and modularity. He's in a delivery sprint so I'll postponed the sync. Maybe we should do a community call about this, I'll schedule with lorenzo in two weeks (I'm in a work trip to Berlin next week).

**Reply by BugliL** [reply]
I'm using PIP and Venv too but I always wanted to change to poetry and we can do it to make easier with deployments.
